### PR TITLE
CAT-899 Consistency if-brick

### PR DESCRIPTION
**Pull request by LauraBebek**
changed default value in if-brick(true) and make it consistent

### CODE REVIEW

File: catroid/src/org/catrobat/catroid/content/bricks/IfLogicBeginBrick.java [hunk 1]
```diff
@@ -169,7 +171,8 @@ public View getViewWithAlpha(int alphaValue) {
 	public View getPrototypeView(Context context) {
 		View prototypeView = View.inflate(context, R.layout.brick_if_begin_if, null);
 		TextView textIfBegin = (TextView) prototypeView.findViewById(R.id.brick_if_begin_prototype_text_view);
-		textIfBegin.setText(String.valueOf(BrickValues.IF_CONDITION));
+		textIfBegin.setText(R.string.brick_if_default_value);
+		Log.d("Lausi", "PRototypeView");
```
Comment (thmq) [inline catroid/src/org/catrobat/catroid/content/bricks/IfLogicBeginBrick.java]: please remove this most unnecessary log message. ("Lausi")

File: catroid/src/org/catrobat/catroid/common/BrickValues.java [hunk 2]
```diff
@@ -52,8 +52,6 @@
 	//Constants Control
 	public static final int WAIT = 1000;
 	public static final int REPEAT = 10;
-	public static final int IF_CONDITION = 1;
-	public static final String NOTE = "add comment here…";
```
Comment (thmq) [inline catroid/src/org/catrobat/catroid/common/BrickValues.java]: please check if really no longer necessary.

Comment (LauraBebek) [inline catroid/src/org/catrobat/catroid/common/BrickValues.java]: it's unused, so we don need it anymore :)

File: catroid/src/org/catrobat/catroid/content/actions/IfLogicAction.java [hunk 3]
```diff
@@ -48,7 +49,12 @@ protected void begin() {
 				return;
 			}
 			Double interpretation = ifCondition.interpretDouble(sprite);
-			ifConditionValue = interpretation.intValue() != 0 ? true : false;
+			String interpretationString = ifCondition.interpretString(sprite);
+			if (interpretationString.equals(R.string.brick_if_default_value)) {
+				ifConditionValue = true;
+			} else {
+				ifConditionValue = interpretation.intValue() != 0 ? true : false;
+			}
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/actions/IfLogicAction.java]: Please always take a look at the ["Intention action icons"](https://www.jetbrains.com/idea/help/intention-actions.html) (yellow light bulb and highlighted code) of your written code. It automatically simplifies your code to
```ifConditionValue = interpretationString.equals(R.string.brick_if_default_value) || (interpretation.intValue() != 0);```

However, `R.string.brick_if_default_value` is an int. So you're comparing a String with an int which will always be false.

File: catroidTest/src/org/catrobat/catroid/uitest/content/brick/BrickValueParameterTest.java [hunk 4]
```diff
@@ -361,8 +361,8 @@ public void testControlBricksDefaultValues() {
 			solo.scrollDownList(fragmentListView);
 		}
 		TextView ifLogicBeginView = (TextView) solo.getView(R.id.brick_if_begin_prototype_text_view);
-		int ifLogicBeginPrototypeValue = Integer.parseInt(ifLogicBeginView.getText().toString());
-		assertEquals("Value in If Begin is not correct", BrickValues.IF_CONDITION, ifLogicBeginPrototypeValue);
+		String ifLogicBeginPrototypeValue = ifLogicBeginView.getText().toString();
+		assertEquals("Value in If Begin is not correct", R.string.brick_if_default_value, ifLogicBeginPrototypeValue);
```
Comment (robertpainsi) [inline catroidTest/src/org/catrobat/catroid/uitest/content/brick/BrickValueParameterTest.java]: int != String
```java
junit.framework.AssertionFailedError: Value in If Begin is not correct expected:<2131624019> but was:<true>
```

File: catroid/src/org/catrobat/catroid/content/bricks/IfLogicBeginBrick.java [hunk 5]
```diff
@@ -56,6 +55,9 @@
 	public IfLogicBeginBrick() {
 		addAllowedBrickField(BrickField.IF_CONDITION);
 	}
+	public IfLogicBeginBrick(String condition) {
```
Comment (robertpainsi) [inline catroid/src/org/catrobat/catroid/content/bricks/IfLogicBeginBrick.java]: Uh, Checkstyle misses that. :wink:

**Comment by robertpainsi** [comment]
I got curious about this PR because I would love to see a single PR fixing all default values and inconsistencies. See [CAT-89](https://jira.catrob.at/browse/CAT-89) and [CAT-1393](https://jira.catrob.at/browse/CAT-1393)

**Comment by thmq** [comment]
reverted the changes, this needs to be re-openend.

**Pull request merged**